python/ray/data/_internal/execution/operators/map_transformer.py

```diff
@@ -151,105 +150,158 @@ def create_map_transformer_from_block_fn(
     )
 
 
-# Below are util functions for converting input/output data.
-
-
-def _input_blocks_to_rows(blocks: Iterable[Block], _: TaskContext) -> Iterable[Row]:
-    """Converts input blocks to rows."""
-    for block in blocks:
-        block = BlockAccessor.for_block(block)
-        for row in block.iter_rows(public_row_format=True):
-            yield row
-
-
-input_blocks_to_rows = MapTransformFn(
-    _input_blocks_to_rows,
-    MapTransformFnDataType.Block,
-    MapTransformFnDataType.Row,
-)
-
-
-def input_blocks_to_batches(
-    blocks: Iterable[Block],
-    _: TaskContext,
-    batch_size: Optional[int] = None,
-    batch_format: str = "default",
-    zero_copy_batch: bool = False,
-) -> Iterable[DataBatch]:
-    """Converts input blocks to batches."""
-    block_iter = iter(blocks)
-    first = next(block_iter, None)
-    if first is None:
-        return []
-    blocks = itertools.chain([first], block_iter)
-    empty_block = BlockAccessor.for_block(first).builder().build()
-    # Don't hold the first block in memory, so we reset the reference.
-    first = None
-
-    # Ensure that zero-copy batch views are copied so mutating UDFs don't error.
-    ensure_copy = not zero_copy_batch and batch_size is not None
-    formatted_batch_iter = batch_blocks(
-        blocks=blocks,
-        stats=None,
-        batch_size=batch_size,
-        batch_format=batch_format,
-        ensure_copy=ensure_copy,
-    )
+# Below are util `MapTransformFn`s for converting input/output data.
 
-    first = next(formatted_batch_iter, None)
-    if first is None:
-        # If the input blocks are all empty, then yield an empty block with same
-        # format as the input blocks.
-        return [empty_block]
-    else:
-        return itertools.chain([first], formatted_batch_iter)
-
-
-def _to_output_blocks(
-    iter: Iterable[MapTransformFnData],
-    _: TaskContext,
-    iter_type: MapTransformFnDataType,
-) -> Iterable[Block]:
-    """Convert UDF-returned data to output blocks.
-
-    Args:
-        iter: the iterable of UDF-returned data, whose type must be one
-            of MapTransformFnDataType.
-        iter_type: the type of the iterable, must match the type of iter.
-    """
-    output_buffer = BlockOutputBuffer(
-        None, DataContext.get_current().target_max_block_size
-    )
-    if iter_type == MapTransformFnDataType.Block:
-        add_fn = output_buffer.add_block
-    elif iter_type == MapTransformFnDataType.Batch:
-        add_fn = output_buffer.add_batch
-    else:
-        assert iter_type == MapTransformFnDataType.Row
-        add_fn = output_buffer.add
-    for data in iter:
-        add_fn(data)
+
+class InputBlocksToRowsMapTransformFn(MapTransformFn):
```

Comment:
it can wait for followup PR, but this naming is pretty long now. I feel a `BlocksToRowsFn` should serve the purpose.

Reply:
I actually intentionally used this long name to make it clear that this class is a subclass of MapTransformFn. Considering this class won't be widely used (currently only `plan_udf_map_op.py`), I think having a lengthy name is fine.

Reply:
Dropped the "input" part.